src/linelast_solver.cpp

```diff
@@ -454,18 +461,17 @@ void LinElastSolver::SetParameterizedProblem(ParameterizedProblem *problem)
    }
 
    // Set BCs, the switch on BC type is done inside SetupRHSBCOperators
-   for (int b = 0; b < global_bdr_attributes.Size(); b++)
+   for (int b = 0; b < problem->battr.Size(); b++)
    {
-      int ti = problem->battr.Find(global_bdr_attributes[b]);
-
-      if (ti >=0)
+      switch (problem->bdr_type[b])
       {
-         type_idx[b] = (LinElastProblem::BoundaryType)problem->bdr_type[ti];
-         if (problem->bdr_type[ti] == LinElastProblem::BoundaryType::DIRICHLET || problem->bdr_type[ti] == LinElastProblem::BoundaryType::NEUMANN)
-         {
-            assert(problem->vector_bdr_ptr[ti]);
-            AddBCFunction(*(problem->vector_bdr_ptr[ti]), problem->battr[ti]);
-         }
+         case BoundaryType::DIRICHLET:
+         case BoundaryType::NEUMANN:
+            assert(problem->vector_bdr_ptr[b]);
```

Comment:
Dirichlet and Neumann should have the same behavior. Does this syntax mean this?
Right now the assertion fails when the boundary type is Neumann but the function is `NULL`